heap_sort/heap_sort.py

```diff
@@ -0,0 +1,135 @@
+"""
+Following is the implementation of heap_sort algorithm.
+It will sort the input array with time complexity O(N*logN)
+N being the size of array.
+Space complexity is O(1).
+"""
+
+
+def left_child_index(i):
+    """
+    :param i: int
+        Index of node in array (that is organized as heap)
+    :return: int
+        Position in array of left child of node
+    """
+
+    return 2 * (i + 1) - 1
+
+
+def right_child_index(i):
+    """
+    :param i: int
+        Index of node in array (that is organized as heap)
+    :return: int
+        Position in array of right child of node
+    """
+
+    return left_child_index(i) + 1
+
+
+def get_children_of_node(a, i):
+    """
+    :param a: list
+        List organized as heap, i.e for each node i:
+            left children is at position 2(i + 1) - 1
+            right children is at position 2(i + 1))
+    :param i: int
+        Index of parent node in array (that is organized as heap)
+    :return: (left children index, left children node) and
+        (right children index, right children node)
+    """
+
+    l = left_child_index(i)  # index and value of children
+    l_val = a[l] if l in range(len(a)) else None
```

Comment:
You can simply use `0 <= l < len(a)`. Any specific reason behind using the generator here?